code/game/objects/items/devices/scanners.dm

```diff
@@ -867,3 +867,115 @@ REAGENT SCANNER
 		dat += "<font color='red'>Retinal misalignment detected.</font><BR>"
 
 	return dat
+
+// Randomly polls tiles on the Z level to give a vague indication if whether something is about.
+/obj/item/space_scanner
+	name = "gravimetric analyzer"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "forensic2"
+	desc = "A handheld suite of sensors designed to scan local areas of space for gravimetric signatures. The compact design renders it somewhat unreliable."
+	slot_flags = SLOT_BELT
+	throwforce = 3
+	w_class = WEIGHT_CLASS_SMALL
+	throw_speed = 5
+	throw_range = 10
+	origin_tech = "materials=4;programming=4;magnets=4"
+	var/obj/item/stock_parts/scanning_module/s_module
+	var/next_use = 0
+	var/scan_cd = 10 SECONDS //how long before we can scan again
+	var/scan_tile_count // The number of tiles to poll randomly.
+
+/obj/item/space_scanner/Initialize()
+	s_module = new(null)
+	update_scanner()
+
+/obj/item/space_scanner/proc/update_scanner()
+	scan_tile_count = 20 + (s_module.rating * 20)
+
+/obj/item/space_scanner/screwdriver_act(mob/living/user, obj/item/I)
+	to_chat(user, "<span class='notice'>You unscrew and remove [s_module] from [src].</span>")
+	s_module.loc = get_turf(loc)
+	s_module = null
+	return
+
+/obj/item/space_scanner/attackby(obj/item/W, mob/user, params)
+	if(!istype(W, /obj/item/stock_parts/scanning_module))
+		return
+	if(!s_module)
+		user.drop_item()
+		W.loc = src
+		s_module = W
+		to_chat(user, "<span class='notice'>You install [s_module] in [src].</span>")
+		update_scanner()
+	else
+		to_chat(user, "<span class='notice'>[src] already has a scanner module.</span>")
+
+/obj/item/space_scanner/attack_self(mob/user as mob)
+	if(!s_module)
+		to_chat(user, "<span class='warning'>An error light blinks. Seems like [src] has no scanner module.</span>")
+		return
+	if(next_use > world.time)
+		to_chat(user, "<span class='warning'>An error light blinks. [src] is still recharging.</span>")
+		return
+	if(has_gravity(get_turf(src)))
+		to_chat(user, "<span class='warning'>An error light blinks. You cannot use [src] inside a gravity well.</span>")
+		return
+
+	var/tiles_scanned = 0
+	var/turf/scanned_tile
+	var/north_count = 0
+	var/south_count = 0
+	var/east_count = 0
+	var/west_count = 0
+	var/magnitude
+	var/turf/U = get_turf(user)
+	var/z_level = U.z
+
+	while(tiles_scanned < scan_tile_count)
+		scanned_tile = locate(rand((2*TRANSITIONEDGE), world.maxx - (2*TRANSITIONEDGE)), rand((2*TRANSITIONEDGE), world.maxy - (2*TRANSITIONEDGE)), z_level)
+		if(!isspaceturf(scanned_tile) || scanned_tile.contents.len > 0)
+			var/direction = get_dir(src, scanned_tile)
+			if(direction & NORTH)
+				north_count++
+			if(direction & SOUTH)
+				south_count++
+			if(direction & EAST)
+				east_count++
+			if(direction & WEST)
+				west_count++
+		tiles_scanned++
+
+	playsound(get_turf(src), 'sound/items/posiping.ogg', 80, 0)
```

Comment:
Can cache the turf to be a bit better on the performance. You did this already on line 931 but line 920 also makes use of the turf.